src/auspex/instruments/bnc.py

```diff
@@ -7,42 +7,72 @@
 #    http://www.apache.org/licenses/LICENSE-2.0
 
 from .instrument import SCPIInstrument, StringCommand, FloatCommand, IntCommand
+from auspex.log import logger
 import time
 import numpy as np
 
 class BNC845(SCPIInstrument):
-    """For controlling the BNC845 microwave source"""
+    """SCPI instrument driver for Berkely Nucleonics BNC845-M RF Signal Generator.
+
+    Properties:
+        frequency: Set the RF generator frequency, in Hz. 0.01-20 GHz.
+        power: Set the RF generator output power, in dBm. No effect, BNC845M output is always +16dBm.
+        output: Toggle RF signal output on/off.
+        pulse: Toggle RF pulsed mode on/off.
+        alc: Toggle source Auto Leveling on/off.
+        mod: Toggle amplitude modulation on/off.
+        pulse_source: Set pulse trigger to INTERNAL or EXTERNAL.
+        freq_source: Set frequency source to INTERNAL or EXTERNAL.
+    """
 
     frequency = FloatCommand(scpi_string="SOURCE:FREQUENCY:FIXED")
     power     = FloatCommand(scpi_string="SOURCE:POWER:LEVEL:IMMEDIATE:AMPLITUDE")
-
-    output    = StringCommand(scpi_string="SOURCE:ROSC:OUTPUT:STATE",value_map={True: 'ON', False: 'OFF'})
-    pulse     = StringCommand(scpi_string="PULSE", value_map={True: 'ON', False: 'OFF'})
-    mod       = StringCommand(scpi_string="MOD", value_map={True: 'ON', False: 'OFF'})
-    alc       = StringCommand(scpi_string="SOURCE:POWER:ALC ", value_map={True: 'ON', False: 'OFF'})
-
-    pulse_source = StringCommand(scpi_string=":PULSE:SOUR", 
+    output    = StringCommand(scpi_string="SOURCE:ROSC:OUTPUT:STATE",value_map={True: '0', False: '1'})
+    pulse     = StringCommand(scpi_string="PULSE", value_map={True: '0', False: '1'})
+    mod       = StringCommand(scpi_string="MOD", value_map={True: '0', False: '1'})
+    alc       = StringCommand(scpi_string="SOURCE:POWER:ALC ", value_map={True: '0', False: '1'})
+    pulse_source = StringCommand(scpi_string=":PULSE:SOUR",
                           value_map={'INTERNAL': 'INT', 'EXTERNAL': 'EXT'})
     freq_source  = StringCommand(scpi_string=":FREQ:SOUR",
                           value_map={'INTERNAL': 'INT', 'EXTERNAL': 'EXT'})
 
     def __init__(self, resource_name, *args, **kwargs):
+        """Berkely Nucleonics BNC845-M RF Signal Generator
```

Comment:
We've been slowly switching to resource_name=None as a default keyword argument to both __init__ and connect, with the latter function using any previously defined self.resource_name.

ExpFactory instantiates with __init__("resource_name") and then just calls connect(), though you're free to run __init() and then connect("resource_name"). Some of the more commonly used instruments should have this pattern. I don't like the extra code to do this, but it makes manual experiment definition very tidy...